### ISSUE TITLE
Tunning N-Layed Neural Network with balanced dataset

### ISSUE THREAD

**Issue by ravyg** [ravyg/its520_project#27]
**This task explains our various attempts of understanding and tuning neural networks**

### 4 Layer Neural Network
- Running with 20 neurons in Layer-1 as **[A,20,16,8,6,B]**
_Deep Neural Net - 4 hidden layers with balanced set of 21550 a subset of our 67059 Tweets_ Feature Vectors.
```python
n_epochs = 2000
batch_size = 1000

def inference_DeepNet4layers(x_tf, A, B):
    # First Hidder Layer 1
    with tf.variable_scope("hidden_1"):
        hidden_1 = layer(x_tf, [A, 20],[20])
    # Inner Layer 2    
    with tf.variable_scope("hidden_2"):
        hidden_2 = layer(hidden_1, [20, 16],[16])
    # Inner Layer 3
    with tf.variable_scope("hidden_3"):
        hidden_3 = layer(hidden_2, [16, 8],[8])
    # Inner Layer 4
    with tf.variable_scope("hidden_4"):
        hidden_4 = layer(hidden_3, [8, 6],[6])
    # Final output Layer 
    with tf.variable_scope("output"):
        output = layer(hidden_4, [6, B], [B])
    return output
```
![screen shot 2016-12-04 at 9 51 13 pm](https://cloud.githubusercontent.com/assets/1249023/20873376/5e8b0b92-ba6c-11e6-8da3-c5f61a05fe05.png)
Also ran this 2000 epochs results remain almost consistent.

- Running with 32 neurons in layer-1 as **[A,32,16,8,6,4,B]**
``` python
def inference_DeepNet2layers(x_tf, A, B):
    # First Hidder Layer 1
    with tf.variable_scope("hidden_1"):
        hidden_1 = layer(x_tf, [A, 32],[32])
    # Inner Layer 2    
    with tf.variable_scope("hidden_2"):
        hidden_2 = layer(hidden_1, [32, 16],[16])
    # Inner Layer 3
    with tf.variable_scope("hidden_3"):
        hidden_3 = layer(hidden_2, [16, 8],[8])
    # Inner Layer 4
    with tf.variable_scope("hidden_4"):
        hidden_4 = layer(hidden_3, [8, 4],[4])
    # Final output Layer 
    with tf.variable_scope("output"):
        output = layer(hidden_4, [4, B], [B])
    return output
```
![screen shot 2016-12-04 at 11 22 22 pm](https://cloud.githubusercontent.com/assets/1249023/20874606/27d01482-ba79-11e6-9caa-3ef91b71915b.png)

### 5 Layer Neural Network
- Running with 64 neurons in layer-1 and symmetric reduction till layer-5 **[A,64,32,16,8,4,B]**
```python
def inference_DeepNet2layers(x_tf, A, B):
    # First Hidder Layer 1
    with tf.variable_scope("hidden_1"):
        hidden_1 = layer(x_tf, [A, 64],[64])
    # Inner Layer 2    
    with tf.variable_scope("hidden_2"):
        hidden_2 = layer(hidden_1, [64, 32],[32])
    # Inner Layer 3
    with tf.variable_scope("hidden_3"):
        hidden_3 = layer(hidden_2, [32, 16],[16])
    # Inner Layer 4
    with tf.variable_scope("hidden_4"):
        hidden_4 = layer(hidden_3, [16, 8],[8])
    # Inner Layer 5    
    with tf.variable_scope("hidden_5"):
        hidden_5 = layer(hidden_4, [8, 4],[4])    
    # Final output Layer 
    with tf.variable_scope("output"):
        output = layer(hidden_5, [4, B], [B])
    return output
```
![screen shot 2016-12-05 at 12 01 16 am](https://cloud.githubusercontent.com/assets/1249023/20875148/fa77e212-ba7d-11e6-979d-022637057ec1.png)

Here we noticed that the graph gave same precision, accuracy and f-measure. But it trained far more quickly.

### Testing 5000 epochs training with 5-Layers [A,64,32,16,8,4,B] set
![screen shot 2016-12-05 at 1 27 17 am](https://cloud.githubusercontent.com/assets/1249023/20876814/25146480-ba8a-11e6-81d7-f91b7e5263dd.png)
